store/mockstore/unistore/cophandler/mpp_exec.go

```diff
@@ -345,6 +345,25 @@ type joinExec struct {
 	inited       bool
 }
 
+func (e *joinExec) getHashKey(keyCol types.Datum, fieldType *types.FieldType) (str string, err error) {
```

Comment:
Why should we do this convert?

Reply:
if not, 1 != decimal(1.0)

Reply:
Why not planner convert types in advance?

Reply:
updated, introduce a commen type to align join key types when get hash keys of string type.